src/olympia/devhub/tests/test_views_edit.py

```diff
@@ -742,6 +742,17 @@ class TestEditDescribeUnlisted(BaseTestEditDescribe, L10nTestsMixin):
     listed = False
     __test__ = True
 
+    def test_no_edit_if_disabled(self):
+        pass
```

Comment:
hmm?   Why is this overridden?

Reply:
mmm I think it's a test I noticed could be useful while I was looking at those tests, I intended to implement it but forgot. I'll look into it.

Reply:
I moved the test up in d2f1a0cbc546d2a3676fa050732393b1746044c8 and clarified the comment in 135528f68f19450e61d49a9cf2fba6f5e9b21cb2. Now I remember why I added that test: the behavior is slightly surprising. If the add-on is pure unlisted, mozilla-disabling it completely hides it from developers, they get a 404. If it had listed versions, they can see it (but of course can't edit it).

This is because of https://github.com/diox/olympia/blob/d2f1a0cbc546d2a3676fa050732393b1746044c8/src/olympia/addons/decorators.py#L35-L44 : they fail the `has_listed_version` part, but *also* the `acl.author_or_unlisted_viewer_or_reviewer()`, despite being an author, because inside that function, we call `check_addon_ownership()` without `allow_mozilla_disabled_addon=True`, so despite being an author, they are denied access.

As far as I can tell, this is intentional, just maybe a little bit surprising.